### PR TITLE
OAuth login for Ripple Trade

### PR DESCRIPTION
**Pull request by h0vhannes**

### CODE REVIEW

**Comment by h0vhannes** [comment]
Guys, please don't merge this, work is still in progress, just made a pull request so you can check it out

File: src/js/services/blob.service.js [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+      }
+      throw new Error('Invalid tilde escape: ' + m);
+    });
+  };
+
+  /**
+   * Sumbit update to blob vault
+   */
+
+  BlobObj.prototype.postUpdate = function(fn) {
```
Comment (vhpoet) [inline src/js/services/blob.service.js]: Please rename `fn` to `callback`, that's what we use every other place.

File: src/js/services/blob.service.js [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+
+    // Callback is optional
+    if (typeof fn !== 'function') {
```
Comment (vhpoet) [inline src/js/services/blob.service.js]: Shorter version would be 
`if ("function" !== typeof callback) callback = $.noop;`

File: src/js/services/id.service.js [hunk 3]
```diff
@@ -10,776 +12,497 @@ var util = require('util'),
...
+        }
+        return address;
+      }
+      return this.resolvedNames[address];
+    };
+
+    /**
+     *
+     */
+    Id.prototype.addressDontHaveName = function(address) {
```
Comment (vhpoet) [inline src/js/services/id.service.js]: Something's wrong with this.

**Comment by vhpoet** [comment]
@h0vhannes the tests are failing

**Comment by h0vhannes** [comment]
Comments fixed, tests issue is in progress

**Comment by h0vhannes** [comment]
Tests issue solved

**Comment by vhpoet** [comment]
LGTM

**Comment by mrajvanshy** [comment]
LGTM

**Comment by vhpoet** [comment]
LGTM

**Pull request merged**